dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/table/DatePeriodResourceTable.java

```diff
@@ -93,10 +93,8 @@ public Optional<List<Object[]>> getPopulateTempTableContent()
 
         List<Object[]> batchArgs = new ArrayList<>();
 
-        // TODO Create a dynamic solution instead of having fixed dates
-
-        Date startDate = new Cal( FIRST_YEAR_SUPPORTED, 1, 1, true ).time();
-        Date endDate = new Cal( LATEST_YEAR_SUPPORTED + 1, 1, 1, true ).time();
+        Date startDate = new Cal( objects.get( 0 ), 1, 1, true ).time();
```

Comment:
We could create variables here as well, to make it more explicit, ie:
``` 
Date startDate = new Cal( objects.get( FIRST_YEAR ), 1, 1, true ).time();
Date endDate = new Cal( objects.get( LAST_YEAR ), 1, 1, true ).time();
```

Reply:
ok, can add +1. I thought the extra 5 years covers for that.